### PR TITLE
KAFKA-5980: FailOnInvalidTimestamp does not log error

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
Call for review @guozhangwang @dguy @bbejeck 

Should be back port this to `0.11.0` and `0.10.2` ? (Not sure if there will be another bug fix release for those...)

File: streams/src/main/java/org/apache/kafka/streams/processor/FailOnInvalidTimestamp.java [hunk 1]
```diff
@@ -45,6 +47,7 @@
  */
 @InterfaceStability.Evolving
 public class FailOnInvalidTimestamp extends ExtractRecordMetadataTimestamp {
+    private static final Logger log = LoggerFactory.getLogger(LogAndSkipOnInvalidTimestamp.class);
```
Comment (tedyu) [inline streams/src/main/java/org/apache/kafka/streams/processor/FailOnInvalidTimestamp.java]: this should be FailOnInvalidTimestamp.class, right ?

File: streams/src/main/java/org/apache/kafka/streams/processor/FailOnInvalidTimestamp.java [hunk 2]
```diff
@@ -60,9 +63,14 @@ public long onInvalidTimestamp(final ConsumerRecord<Object, Object> record,
                                    final long recordTimestamp,
                                    final long previousTimestamp)
             throws StreamsException {
+        log.error("Input record {} has invalid (negative) timestamp. " +
```
Comment (dbolshak) [inline streams/src/main/java/org/apache/kafka/streams/processor/FailOnInvalidTimestamp.java]: your message for the logger and for the exception, could you please extract it to `val`?

**Comment by mjsax** [comment]
Updates this. Thanks for the review @tedyu @dbolshak

**Review by dbolshak — COMMENTED** [review]
LGTM, need to ask committers to take a look.

**Review by dguy — APPROVED** [review]
Thanks @mjsax, LGTM

**Comment by guozhangwang** [comment]
Merged to trunk.

**Pull request closed without merging**